src/BuildScriptGenerator/Node/NodeVersionChecker.cs

```diff
@@ -0,0 +1,36 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Oryx.BuildScriptGenerator.Node
+{
+    [Checker(NodeConstants.NodeJsName)]
+    public class NodeVersionChecker : IChecker
+    {
+        public IEnumerable<ICheckerMessage> CheckToolVersions(IDictionary<string, string> tools)
+        {
+            if (tools.ContainsKey(NodeConstants.NodeJsName))
+            {
+                var used = tools[NodeConstants.NodeJsName];
+                if (SemanticVersionResolver.CompareVersions(used, NodeScriptGeneratorOptionsSetup.NodeLtsVersion) < 0)
+                {
+                    return new[]
+                    {
+                        new CheckerMessage($"An outdated version of Node.js was used ({used}). Consider updating.")
```

Comment:
👍 nice check...but I would like the message to be more clear...like saying current LTS version is this, but you are currently using this, so consider updating etc... (should 'was used' be 'is being used'?)

Reply:
I didn't want to say "LTS" or "current" because these definition change in Oryx on a different cadence than how they change in Node.js. I can't think of a way to resolve this other than adding a link to the Oryx docs.